src/vcpkg/versions.cpp

```diff
@@ -50,93 +51,185 @@ namespace vcpkg::Versions
         return hash<string>()(key.port_name) ^ (hash<string>()(key.version.to_string()) >> 1);
     }
 
-    ExpectedS<RelaxedVersion> RelaxedVersion::from_string(const std::string& str)
+    // 0|[1-9][0-9]*
+    static const char* parse_skip_number(const char* const s, uint64_t* const n)
```

Comment:
I observe all places calling this do something of the form:

```
x = fn(x, ...)
```

so you may want to consider making the first param an in_out parameter.

Reply:
I felt this was cleaner due to reducing indirection (ret value + in param + out param instead of ret value + in_out param)